pandas/tests/test_frame.py

```diff
@@ -12328,6 +12328,24 @@ def test_unstack_dtypes(self):
         expected = Series({'float64' : 2, 'object' : 2})
         assert_series_equal(result, expected)
 
+        # GH7405
+        for c, d in (np.zeros(5), np.zeros(5)), \
```

Comment:
i do not think `np.zeros` requires dtypes here, because `left` and `right` frame are both defined off the `df` frame here. i would rather keep this replicate of https://github.com/pydata/pandas/issues/7405. also, in the 2nd iteration of for loop there is `np.arange` with explicit data type.